packages/query/src/index.ts

```diff
@@ -91,4 +119,58 @@ export class LiveQuery extends TxProcessor implements Client {
     await this.client.tx(tx)
     await super.tx(tx)
   }
+
+  async updateDoc (updatedDoc: Doc, tx: TxUpdateDoc<Doc>): Promise<void> {
+    const ops = tx.operations as any
+    for (const key in ops) {
+      if (key.startsWith('$')) {
+        const operator = getOperator(key)
+        operator(updatedDoc, ops[key])
+      } else {
+        (updatedDoc as any)[key] = ops[key]
+      }
+    }
+    updatedDoc.modifiedBy = tx.modifiedBy
+    updatedDoc.modifiedOn = tx.modifiedOn
+  }
+
+  private sort (q: Query, tx: TxUpdateDoc<Doc>): void {
+    const sort = q.options?.sort
+    if (sort === undefined) return
+    let needSort = sort.modifiedBy !== undefined || sort.modifiedOn !== undefined
+    if (!needSort) needSort = this.checkNeedSort(sort, tx)
+
+    if (needSort) resultSort(q.result as Doc[], sort)
+  }
+
+  private checkNeedSort (sort: SortingQuery<Doc>, tx: TxUpdateDoc<Doc>): boolean {
+    const ops = tx.operations as any
+    for (const key in ops) {
+      if (key.startsWith('$')) {
+        for (const opKey in ops[key]) {
+          if (opKey in sort) return true
+        }
+      } else {
+        if (key in sort) return true
+      }
+    }
+    return false
+  }
+
+  async callback (updatedDoc: Doc, q: Query): Promise<void> {
+    q.result = q.result as Doc[]
+
+    if (q.options?.limit !== undefined && q.result.length > q.options.limit) {
+      if (q.result[q.options?.limit]._id === updatedDoc._id) {
+        const res = await this.findAll(q._class, q.query, q.options)
+        q.result = res
+        q.total = res.total
+        q.callback(res)
+        return
+      }
+      if (q.result.pop()?._id !== updatedDoc._id) q.callback(Object.assign(q.result, { total: q.total }))
```

Comment:
You you please comment on this case?

Reply:
Checking whether the updated document is within the limit